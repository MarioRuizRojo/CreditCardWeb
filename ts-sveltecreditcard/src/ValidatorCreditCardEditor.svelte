<script lang="ts">
  import { _ } from "svelte-i18n";

  import { MyValidator } from "./MyValidator";
  import { MyValidation } from "./MyValidation";

  /**
   * MyValidator instance to validate all fields in CreditCardEditor svelte component
   */
  export let validator: MyValidator;

  /**
   * MyValidation list instances that define the rule and error message for each field in CreditCardEditor svelte component
   */
  let validations: MyValidation[];

  /**
   * validation error message required
   */
  let errorRequired: string = $_("required");
  /**
   * validation error message for only letters rule
   */
  let errorOnlyLetters: string = $_("onlyletters");
  /**
   * validation error message for 16 digits rule
   */
  let error16dig: string = $_("should16dig");
  /**
   * validation error message for date format rule
   */
  let errorDateForm: string = $_("shoulddateform");
  /**
   * validation error message for 3 digits rule
   */
  let error3dig: string = $_("should3dig");

  /**
   * it creates the validations with the error messages in the selected language
   */
  function updateLangValidations(): void {
    validations = Array(4);
    validations[0] = new MyValidation(
      errorOnlyLetters,
      new RegExp(/([a-zA-Z]+)/i)
    );
    validations[1] = new MyValidation(error16dig, new RegExp(/([0-9]{16})/i));
    validations[2] = new MyValidation(
      errorDateForm,
      new RegExp(/^(0[1-9]|1[0-2])\/([0-9]{2})$/i)
    );
    validations[3] = new MyValidation(error3dig, new RegExp(/([0-9]{3})/i));
    validator = new MyValidator(validations, errorRequired);
  }

  //error messages update when lang changes
  $: {
    errorRequired = $_("required");
    errorOnlyLetters = $_("onlyletters");
    error16dig = $_("should16dig");
    errorDateForm = $_("shoulddateform");
    error3dig = $_("should3dig");
    updateLangValidations();
  }

  updateLangValidations();
  validator = new MyValidator(validations, errorRequired);
</script>
