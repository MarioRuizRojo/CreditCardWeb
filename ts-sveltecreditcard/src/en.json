{
    "addcard": "Add Credit Card",
    "loading": "Loading...",
    "placeholdercardholdername": "Card Holder Name",
    "placeholdercardsnumber": "Card's Number",
    "placeholderdate": "MM/YY",
    "placeholdercvv": "CVV",
    "add": "Add",
    "required": "Required",
    "onlyletters": "Should contain only letters",
    "should16dig": "Should contain 16 digits",
    "shoulddateform": "Should have date format",
    "should3dig": "Should contain 3 digits",
    "savechanges": "Save Changes",
    "creditcards": "Credit Cards",
    "creditcardapp": "Credit Cards App",
    "selectlang": "Select language",
    "editcard": "Edit Credit Card"
}