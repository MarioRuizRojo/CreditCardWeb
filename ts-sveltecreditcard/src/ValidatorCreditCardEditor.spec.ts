import ValidatorCreditCardEditor from './ValidatorCreditCardEditor.svelte';
import { render } from '@testing-library/svelte';

//language support in tests
import { locale, _, addMessages } from 'svelte-i18n';
import en from './en.json';
/**
 * It sets up the language translation engine before running the tests
 */
beforeAll(function (): void {
    addMessages('en', en);
    locale.set('en');
});

/**
 * it expects default values for ValidatorCreditCardEditor attributes
 */
it('check constructor', function (): void {
    const rendered = render(ValidatorCreditCardEditor);
    expect(rendered).toMatchSnapshot();
});
