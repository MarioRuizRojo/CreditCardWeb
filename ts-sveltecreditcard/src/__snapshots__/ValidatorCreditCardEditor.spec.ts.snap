// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check constructor 1`] = `
Object {
  "component": ValidatorCreditCardEditor {
    "$$": Object {
      "after_update": Array [],
      "before_update": Array [],
      "bound": Object {},
      "callbacks": Object {},
      "context": Map {},
      "ctx": Array [
        MyValidator {
          "errorsToShow": Array [
            "",
            "",
            "",
            "",
          ],
          "passed": Array [
            false,
            false,
            false,
            false,
          ],
          "requiredError": "Required",
          "validations": Array [
            MyValidation {
              "errorMessage": "Should contain only letters",
              "ruleFormat": /\\(\\[a-zA-Z\\]\\+\\)/i,
            },
            MyValidation {
              "errorMessage": "Should contain 16 digits",
              "ruleFormat": /\\(\\[0-9\\]\\{16\\}\\)/i,
            },
            MyValidation {
              "errorMessage": "Should have date format",
              "ruleFormat": /\\^\\(0\\[1-9\\]\\|1\\[0-2\\]\\)\\\\/\\(\\[0-9\\]\\{2\\}\\)\\$/i,
            },
            MyValidation {
              "errorMessage": "Should contain 3 digits",
              "ruleFormat": /\\(\\[0-9\\]\\{3\\}\\)/i,
            },
          ],
        },
        [Function],
      ],
      "dirty": Array [
        -1,
      ],
      "fragment": Object {
        "c": [Function],
        "d": [Function],
        "i": [Function],
        "l": [Function],
        "m": [Function],
        "o": [Function],
        "p": [Function],
      },
      "not_equal": [Function],
      "on_destroy": Array [
        [Function],
        [Function],
      ],
      "on_disconnect": Array [],
      "on_mount": Array [],
      "props": Object {
        "validator": 0,
      },
      "skip_bound": false,
      "update": [Function],
    },
    "$$set": [Function],
    "$capture_state": [Function],
    "$inject_state": [Function],
  },
  "container": <body>
    <div />
  </body>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
